server/mdm/maintainedapps/installers.go

```diff
@@ -0,0 +1,98 @@
+package maintainedapps
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+// DownloadInstaller downloads the maintained app installer located at the given URL.
+func DownloadInstaller(ctx context.Context, installerURL string, maxSize int64) ([]byte, string, error) {
+	// validate the URL before doing the request
+	_, err := url.ParseRequestURI(installerURL)
+	if err != nil {
+		return nil, "", fleet.NewInvalidArgumentError(
+			"fleet_maintained_app.url",
+			fmt.Sprintf("Couldn't download maintained app installer. URL (%q) is invalid", installerURL),
+		)
+	}
+
+	client := fleethttp.NewClient(fleethttp.WithTimeout(30 * time.Second))
+	client.Transport = fleethttp.NewSizeLimitTransport(maxSize)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, installerURL, nil)
+	if err != nil {
+		return nil, "", ctxerr.Wrapf(ctx, err, "creating request for URL %s", installerURL)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		var maxBytesErr *http.MaxBytesError
+		if errors.Is(err, fleethttp.ErrMaxSizeExceeded) || errors.As(err, &maxBytesErr) {
+			return nil, "", fleet.NewInvalidArgumentError(
+				"fleet_maintained_app.url",
+				fmt.Sprintf("Couldn't download maintained app installer. URL (%q). The maximum file size is %d MB", installerURL, maxSize/(1024*1024)),
```

Comment:
is this correct? `maxSize` is in GB:

```go
const maxMaintainedInstallerSizeBytes int64 = 1024 * 1024 * 1024 * 3 // 3GB
```

I also wonder if this constant should live in the `maintainedapps` package? not super importatn